tests/expose_test.go

```diff
@@ -460,13 +437,73 @@ var _ = Describe("Expose", func() {
 				Expect(err).ToNot(HaveOccurred())
 
 				By("Waiting for the pod to report a successful connection attempt.")
-				getStatus = func() k8sv1.PodPhase {
-					pod, err := virtClient.CoreV1().Pods(job.Namespace).Get(job.Name, k8smetav1.GetOptions{})
-					Expect(err).ToNot(HaveOccurred())
-					return pod.Status.Phase
+				waitForJobCompletion(&virtClient, job, "success")
+			})
+
+			It("[test_id:343]Should Verify an exposed service of a VM is not functional after VM deletion.", func() {
+				By("Getting back the cluster IP given for the service")
+				svc, err := virtClient.CoreV1().Services(vm.Namespace).Get(serviceName, k8smetav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				serviceIP := svc.Spec.ClusterIP
+
+				By("Starting a pod which tries to reach the VMI via ClusterIP")
+				job := tests.NewHelloWorldJob(serviceIP, servicePort)
+				job, err = virtClient.CoreV1().Pods(vm.Namespace).Create(job)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Waiting for the pod to report a successful connection attempt")
+				waitForJobCompletion(&virtClient, job, "success")
+
+				By("Comparing the service's endpoints IP address to the VM pod IP address.")
+				// Get the IP address of the VM pod.
+				vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(vm.Name, &k8smetav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				vmPod := tests.GetRunningPodByVirtualMachineInstance(vmi, vmi.Namespace)
+				Expect(err).ToNot(HaveOccurred())
+				vmPodIpAddress := vmPod.Status.PodIP
+
+				// Get the IP address of the service's endpoint.
+				endpointsName := serviceName
+				svcEndpoints, err := virtClient.CoreV1().Endpoints(vm.Namespace).Get(endpointsName, k8smetav1.GetOptions{})
+				Expect(svcEndpoints).ToNot(BeNil())
+				endptSubsetIpAddress := func() string {
```

Comment:
No need for this it should be only one address (the virt-launcher pod) if there is more we should fail to test

Reply:
@SchSeba The Subsets member of the Endpoints struct is defined as an array; the Addresses member of the EndpointSubset strucy is also an array.
`Ports []EndpointPort`
`Addresses []EndpointAddress`
Therefore, I must go over all instances of each array, in case the IP address I'm looking for is "hiding" in the non-first subset and/or non-first address.

Reply:
It should be only one because the service point to one pod only.
if there is multiple it should fail (the service point to multiple pods)

Reply:
OK, in that case I agree. I'll fix this to look for one address in one subset only, and fail the test if there is more than one subset.

Reply:
Done.